src/analysis/ptyp_of_type.ml

```diff
@@ -132,17 +132,28 @@ and extension_constructor id {
     ~args:(constructor_arguments ext_args)
     ?res:(Option.map ~f:core_type ext_ret_type)
     (var_of_id id)
-and value_description id { val_type; val_kind=_; val_loc; val_attributes; _ } =
+and const_modalities modalities =
+  modalities
+  |> Mode.Modality.Value.Const.to_list
+  |> List.map ~f:(fun modality ->
+    let modality_str = Format.asprintf "%a" Mode.Modality.print modality in
+    Location.mknoloc @@ Parsetree.Modality modality_str)
+and value_description id { val_type; val_kind=_; val_loc; val_attributes; val_modalities; _ } =
   let type_ = core_type val_type in
   {
     Parsetree.pval_name = var_of_id id;
     pval_type = type_;
     pval_prim = [];
     pval_attributes = val_attributes;
+    pval_modalities = const_modalities @@ Mode.Modality.Value.to_const_exn val_modalities;
```

Comment:
Is it safe to call `to_const_exn` here? It seems like there should be a better way.